File: ruby_SDK/PayHubSDK/lib/PayHubSDK/com/payhub/ws/api/bill_information.rb
class BillInformation
  include JsonSerializer
  ATTRS=[:version,:createdAt,:lastModified,:createdBy,:lastModifiedBy,:metaData]
  attr_accessor :url,:transactionManager,:bill,:errors,*ATTRS

  # @param [Object] transactionManager
  def initialize(transactionManager=nil)
    @transactionManager=transactionManager
    @transactionType=TransactionType::Bill
  end

  def convertData(json)
      obj = JSON.parse(json)
      @version=obj['version']
      @createdAt=obj['createdAt']
      @lastModified=obj['lastModified']
      @createdBy=obj['createdBy']
      @lastModifiedBy=obj['lastModifiedBy']
      @metaData=obj['metaData']
  end

  def convertDataToBill(json)
      @bill = Bill.from_json(json)
  end
  private :convertData, :convertDataToBill
  # @param [Object] sale id
  def getBillForSaleInformationByTransactionId(saleId)
    url = @transactionManager.url+Sale::SALE_ID_LINK+saleId.to_s+"/bill"
    response = @transactionManager.doGet(url, @transactionManager.token)
    return nil if response==nil or response==""
    result=JSON.parse(response)
    errors ||= Array.new
    if result.include?('errors')
      result['errors'].each do |error|
        response_tmp = Errors.from_json(JSON.generate(error))
        errors.push(response_tmp)
      end
      @errors=errors
      return @errors
    end
    convertData(response)
    convertDataToBill(response)
  end

  # @param [Object] bill id
  def getBillForSaleInformationById(id)
    url = @transactionManager.url+"bill-for-sale/"+id.to_s
    response = @transactionManager.doGet(url, @transactionManager.token)
    return nil if response==nil or response==""
    result=JSON.parse(response)
    errors ||= Array.new
    if result.include?('errors')
      result['errors'].each do |error|
        response_tmp = Errors.from_json(JSON.generate(error))
        errors.push(response_tmp)
      end
      @errors=errors
      return @errors
    end
    convertData(response)
    convertDataToBill(response)
  end
  # @param [Object] recurring bill id
  def getBillForRecurringBillInformationByTransactionId(id)
    url = @transactionManager.url+RecurringBill::RECURRENT_BILL_ID_LINK+id.to_s+"/bill"
    response = @transactionManager.doGet(url, @transactionManager.token)
    return nil if response==nil or response==""
    result=JSON.parse(response)
    errors ||= Array.new
    if result.include?('errors')
      result['errors'].each do |error|
        response_tmp = Errors.from_json(JSON.generate(error))
        errors.push(response_tmp)
      end
      @errors=errors
      return @errors
    end
    convertData(response)
    convertDataToBill(response)
  end
  # @param [Object] bill id
  def getBillForRecurringBillInformationById(id)
    url = @transactionManager.url+"bill-for-recurring-bill/"+id.to_s
    response = @transactionManager.doGet(url, @transactionManager.token)
    return nil if response==nil or response==""
    result=JSON.parse(response)
    errors ||= Array.new
    if result.include?('errors')
        result['errors'].each do |error|
          response_tmp = Errors.from_json(JSON.generate(error))
          errors.push(response_tmp)
        end
      @errors=errors
      return @errors
    end
    convertData(response)
    convertDataToBill(response)
  end


end

File: c#_SDK/PayHubSDK/PayHubSDK/com/payhub/ws/api/BillInformation.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using PayHubSDK.com.payhub.ws.api;
using PayHubSDK.com.payhub.ws.model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Runtime.Serialization;
using System.Text;

namespace PayHubSDK.com.payhub.ws.api
{
    [DataContract]
    public class BillInformation
    {
        [DataMember(Name = "version")]
        private string _version;
        [DataMember(Name = "createdAt")]
        private string _createdAt;
       [DataMember(Name = "lastModified")]
        private string _lastModified;
        [DataMember(Name = "createdBy")]
        private string _createdBy;
       [DataMember(Name = "lastModifiedBy")]
        private string _lastModifiedBy;
        [DataMember(Name = "metaData")]
        private Object _metaData;
	    private TransactionManager transactionManager;
        public TransactionManager TransactionManager { set { this.transactionManager = value; } }
	    private TransactionType transactionType;

        public string Version { get { return this._version; } set { this._version = value; } }
        public string CreatedAt { get { return this._createdAt; } set { this._createdAt = value; } }
        public string LastModified { get { return this._lastModified; } set { this._lastModified = value; } }
        public string CreatedBy { get { return this._createdBy; } set { this._createdBy = value; } }
        public string LastModifiedBy { get { return this._lastModifiedBy; } set { this._lastModifiedBy = value; } }
        public Object metaData { get { return this._metaData.ToString(); } set { this._metaData = value.ToString(); } }
        public string url { get; set; }
        public Bill bill { get; set; }

        public BillInformation() {
            this.transactionType = TransactionType.Bill;
        }
        public BillInformation(TransactionManager t)
        {
            this.transactionManager = t;
            this.transactionType = TransactionType.Bill;
        }
        private void convertData(String json)
        {
            // read from file, convert it to user class
            var node = JObject.Parse(json);
            this.Version = (string)node["version"];
            this.CreatedAt = (string)node["createdAt"];
            this.LastModified = (string)node["lastModified"];
            this.CreatedBy = (string)node["createdBy"];
            this.LastModifiedBy = (string)node["lastModifiedBy"];
            this.metaData = (string)node["metaData"];
        }

        private void convertDataToBill(String json){
            bill = JsonConvert.DeserializeObject<Bill>(json, new JsonSerializerSettings { DefaultValueHandling = DefaultValueHandling.Ignore });
	    }

        public void getBillForSaleInformationByTransactionId(String id){
		    String url = this.url+id+"/bill";
		    HttpWebRequest request = this.transactionManager.setHeadersGet(url, this.transactionManager.oauthToken);
		    String json=this.transactionManager.doGet(request);
		    convertData(json);
		    convertDataToBill(json);
	    }
	    public void getBillForSaleInformationById(String id){
		    String url = this.transactionManager.Url+"bill-for-sale/"+id;
		    HttpWebRequest request = this.transactionManager.setHeadersGet(url, this.transactionManager.oauthToken);
		    String json=this.transactionManager.doGet(request);
		    convertData(json);
		    convertDataToBill(json);
	    }
	    public void getBillForRecurringBillInformationByTransactionId(String id){
		    String url = this.url+id+"/bill";
		    HttpWebRequest request = this.transactionManager.setHeadersGet(url, this.transactionManager.oauthToken);
		    String json=this.transactionManager.doGet(request);
		    convertData(json);
		    convertDataToBill(json);
	    }
	    public void getBillForRecurringBillInformationById(String id){
		    String url = this.transactionManager.Url+"bill-for-recurring-bill/"+id;
		    HttpWebRequest request = this.transactionManager.setHeadersGet(url, this.transactionManager.oauthToken);
		    String json=this.transactionManager.doGet(request);
		    convertData(json);
		    convertDataToBill(json);
	    }

    }
}
